src/foam/dao/HTTPSink.java

```diff
@@ -26,17 +29,19 @@
 {
   protected String url_;
   protected String bearerToken_;
+  protected DUGDigestConfig dugDigestConfig_;
   protected Format format_;
   protected PropertyPredicate propertyPredicate_;
   protected boolean outputDefaultValues_;
 
   public HTTPSink(String url, Format format) {
-    this(url, "", format, null, false);
+    this(url, "", null, format, null, false);
   }
 
-  public HTTPSink(String url, String bearerToken, Format format, PropertyPredicate propertyPredicate, boolean outputDefaultValues) {
+  public HTTPSink(String url, String bearerToken, DUGDigestConfig dugDigestConfig, Format format, PropertyPredicate propertyPredicate, boolean outputDefaultValues) {
```

Comment:
Please create a new sink class - HTTPDigestSink or something. The DUG rule below can create whatever it needs.